File: src/main/java/com/flexpoker/signup/query/repository/SignUpRepository.java
package com.flexpoker.signup.query.repository;

import java.util.UUID;

import com.flexpoker.signup.command.aggregate.SignUpUser;

public interface SignUpRepository {

    boolean usernameExists(String username);

    boolean signUpCodeExists(UUID signUpCode);

    UUID findAggregateIdByUsernameAndSignUpCode(String username, UUID signUpCode);

    UUID findSignUpCodeByUsername(String username);

    void storeSignUpInformation(UUID aggregateId, String username, UUID signupCode);

    void storeNewlyConfirmedUsername(String username);

    SignUpUser fetchSignUpUser(UUID signUpUserId);

    void saveSignUpUser(SignUpUser signUpUser);

}

File: src/main/kotlin/com/flexpoker/signup/repository/SignUpRepository.kt
package com.flexpoker.signup.repository

import com.flexpoker.signup.SignUpUser
import java.util.UUID

interface SignUpRepository {
    fun usernameExists(username: String): Boolean
    fun signUpCodeExists(signUpCode: UUID): Boolean
    fun findAggregateIdByUsernameAndSignUpCode(username: String, signUpCode: UUID): UUID?
    fun findSignUpCodeByUsername(username: String): UUID
    fun storeSignUpInformation(aggregateId: UUID, username: String, signUpCode: UUID)
    fun storeNewlyConfirmedUsername(username: String)
    fun fetchSignUpUser(signUpUserId: UUID): SignUpUser?
    fun saveSignUpUser(signUpUser: SignUpUser)
}